src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -750,6 +750,56 @@ private static boolean isVCFStream(final InputStream stream, final String MAGIC_
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
+    private Object genotypeValsToTargetType(final String key, final String vals) {
+        if (!header.hasFormatLine(key)) {
+            return vals;
+        }
+
+        final VCFFormatHeaderLine formLine = header.getFormatHeaderLine(key);
+        final VCFHeaderLineType valType = formLine.getType();
+        final boolean isSingleVal = formLine.isFixedCount()
```

Comment:
this doesn't seem to be a reasonable way to find the count.....I think you should consider the count type (A, R, G, etc) instead.

Reply:
please add a test with a A-type FORMAT field, and a string that contains commas as the value.

Reply:
@yfarjoun, @lbergelson Hello! Thanks for review! 
All requested changes are done